src/App/DocumentObserver.cpp

```diff
@@ -357,6 +446,17 @@ const char *SubObjectT::getElementName() const {
     return Data::findElementName(subname.c_str());
 }
 
+bool SubObjectT::hasSubObject() const
+{
+    return Data::findElementName(subname.c_str()) != subname.c_str();
+}
+
+bool SubObjectT::hasSubElement() const
+{
+    auto element = getElementName();
+    return (element != nullptr) && (element[0] != 0);

```

Comment:
This is the other case where existence of a pointer is expressed in terms of a `nullptr` comparison. I would prefer: `element && element[0] != '\0'`.